source/common/http/matching/inputs.h

```diff
@@ -139,6 +139,42 @@ class HttpResponseTrailersDataInputFactory
 public:
   HttpResponseTrailersDataInputFactory() : HttpHeadersDataInputFactoryBase("response_trailers") {}
 };
+
+class HttpRequestQueryParamsDataInput : public Matcher::DataInput<HttpMatchingData> {
+public:
+  explicit HttpRequestQueryParamsDataInput(const std::string& query_param, bool url_decoded)
+      : query_param_(query_param), url_decoded_(url_decoded) {}
+
+  Matcher::DataInputGetResult get(const HttpMatchingData& /*data*/) const override;
+
+private:
+  const std::string query_param_;
+  bool url_decoded_;
+};
+
+class HttpRequestQueryParamsDataInputFactory : public Matcher::DataInputFactory<HttpMatchingData> {
+public:
+  std::string name() const override { return "query_params"; }
+
+  Matcher::DataInputFactoryCb<HttpMatchingData>
+  createDataInputFactoryCb(const Protobuf::Message& config,
+                           ProtobufMessage::ValidationVisitor& validation_visitor) override {
+    const auto& typed_config = MessageUtil::downcastAndValidate<
+        const envoy::type::matcher::v3::HttpRequestQueryParamMatchInput&>(config,
+                                                                          validation_visitor);
+
+    return [query_param = typed_config.query_param(), url_decoded = typed_config.url_decoded()] {
+      return std::make_unique<HttpRequestQueryParamsDataInput>(query_param, url_decoded);
+    };
+  };
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<envoy::type::matcher::v3::HttpRequestQueryParamMatchInput>();
+  }
+
+private:
+  const std::string name_;
```

Comment:
This is unused since it is hard coded to `query_params` in name() function

Either remove this or set it to `query_params` and use it in name() function.

Reply:
Indeed, sorry about this... Fixed here: https://github.com/envoyproxy/envoy/pull/24627/commits/5305a220ce918ae6fbac237d031bcb97fed2ff13#diff-2df46d0706764a215b8c3bf412d40761a2f7c02f8abb798aa929cad394e721b7L175